src/Symfony/Component/Security/Core/Security.php

```diff
@@ -57,4 +58,23 @@ public function getToken(): ?TokenInterface
     {
         return $this->container->get('security.token_storage')->getToken();
     }
+
+    /**
+     * Logout the current user automatically. Dispatch the logout event.
+     */
+    public function logout(): void
+    {
+        $request = $this->container->get('request_stack')->getCurrentRequest();
+        $logoutEvent = new LogoutEvent($request, $this->container->get('security.token_storage')->getToken());
+
+        $firewallName = $this
+            ->container
+            ->get('security.firewall.map')
+            ->getFirewallConfig($request)
+            ->getName()
+        ;
```

Comment:
Please use a single line for these chained methods (we generally don't wrap long lines in Symfony).

Also, I think `getFirewallConfig()` may return `null`. We should check it before calling `getName()`.

Reply:
Ok thanks. What do we do if `getFirewallConfig()` returns null ? Exit the method and so doing nothing, or throw an exception ?

Reply:
If null is returned, the request is not behind a firewall as far as I can see. So I would say it makes sense to throw an exception like "It is not possible to logout, as the request is not behind a firewall."

Reply:
Fixed